File: backend/models/todo.py
"""
待办清单模型
"""
from datetime import datetime
from sqlalchemy import Column, String, Boolean, DateTime, JSON
from database import Base


class TodoList(Base):
    """待办清单表"""
    __tablename__ = "todo_lists"

    id = Column(String, primary_key=True)
    user_id = Column(String, nullable=False, index=True)
    recipe_names = Column(JSON, default=[])  # ["番茄炒蛋", "紫菜蛋花汤"]
    servings = Column(String, nullable=True)
    status = Column(String(20), default="pending")  # pending, shopping, completed
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)


class TodoItem(Base):
    """待办清单项表"""
    __tablename__ = "todo_items"

    id = Column(String, primary_key=True)
    todo_list_id = Column(String, nullable=False, index=True)
    user_id = Column(String, nullable=False, index=True)
    ingredient_name = Column(String(100), nullable=False)
    quantity = Column(String(50), nullable=False)
    is_purchased = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.utcnow)

File: app/lib/data/models/todo.dart
class TodoItem {
  final String id;
  final String ingredientName;
  final String quantity;
  final bool isPurchased;

  TodoItem({
    required this.id,
    required this.ingredientName,
    required this.quantity,
    this.isPurchased = false,
  });

  factory TodoItem.fromJson(Map<String, dynamic> json) => TodoItem(
        id: json['id'],
        ingredientName: json['ingredient_name'],
        quantity: json['quantity'],
        isPurchased: json['is_purchased'] ?? false,
      );

  TodoItem copyWith({bool? isPurchased}) => TodoItem(
        id: id,
        ingredientName: ingredientName,
        quantity: quantity,
        isPurchased: isPurchased ?? this.isPurchased,
      );
}

class TodoListModel {
  final String id;
  final List<String> recipeNames;
  final String? servings;
  final String status;
  final int totalItems;
  final int pendingItems;

  TodoListModel({
    required this.id,
    required this.recipeNames,
    this.servings,
    required this.status,
    required this.totalItems,
    required this.pendingItems,
  });

  factory TodoListModel.fromJson(Map<String, dynamic> json) => TodoListModel(
        id: json['id'],
        recipeNames: List<String>.from(json['recipe_names'] ?? []),
        servings: json['servings'],
        status: json['status'] ?? 'pending',
        totalItems: json['total_items'] ?? 0,
        pendingItems: json['pending_items'] ?? 0,
      );
}
